src/clarity-angular/datagrid/providers/sort.ts

```diff
@@ -18,9 +18,12 @@ export class Sort {
     public get comparator(): Comparator<any> {
         return this._comparator;
     }
+
     public set comparator(value: Comparator<any>) {
-        this._comparator = value;
-        this.emitChange();
+        if (typeof value !== "undefined") {
+            this._comparator = value;
+            this.emitChange();
+        }
```

Comment:
... not here.

If you're handling undefined user input, do it directly at the input level, not at the deep internal provider level. This change has too many hidden effects (that might not matter right now) for what it solves. The surface solution is safer.